### PR TITLE
Added healthcheck support. Updated entrypoint.sh for mysql 8.0

### PR DESCRIPTION
**Pull request by sergey-zabolotny**

### CODE REVIEW

File: mysql-8.0/docker-entrypoint.sh [hunk 1]
```diff
@@ -59,20 +77,13 @@ _check_config() {
...
 	local conf="$1"; shift
-	"$@" --verbose --help --log-bin-index="$(mktemp -u)" 2>/dev/null | awk '$1 == "'"$conf"'" { print $2; exit }'
+	"$@" --verbose --help --log-bin-index="$(mktemp -u)" 2>/dev/null \
+		| awk '$1 == "'"$conf"'" && /^[^ \t]/ { sub(/^[^ \t]+[ \t]+/, ""); print; exit }'
+	# match "datadir      /some/path with/spaces in/it here" but not "--xyz=abc\n     datadir (xyz)"
 }
 
 # allow the container to be started with `--user`
 if [ "$1" = 'mysqld' -a -z "$wantHelp" -a "$(id -u)" = '0' ]; then
-	# Docksal: copy custom settings (if mounted) from /var/www/.docksal/etc/mysql/my.cnf and fix permissions
```
Comment (lmakarov) [inline mysql-8.0/docker-entrypoint.sh]: This custom block is used to inject Docksal db settings overrides. Please restore.

File: tests/test.bats [hunk 2]
```diff
@@ -32,13 +79,15 @@ teardown () {
 
 @test "Default database present" {
 	[[ $SKIP == 1 ]] && skip
+	_healthcheck_wait
```
Comment (lmakarov) [inline tests/test.bats]: `_healthcheck_wait` is only necessary after `make start`. Please remove it here.

File: tests/test.bats [hunk 3]
```diff
@@ -32,13 +79,15 @@ teardown () {
...
 
 	run make mysql-query QUERY='SHOW DATABASES;'
 	[[ "$output" =~ "default" ]]
 }
 
 @test "Check variables" {
 	[[ $SKIP == 1 ]] && skip
+	_healthcheck_wait
```
Comment (lmakarov) [inline tests/test.bats]: `_healthcheck_wait` is only necessary after `make start`. Please remove it here.

**Review by lmakarov — CHANGES_REQUESTED** [review]
@sergey-zabolotny what was the reason for `mysql-8.0/docker-entrypoint.sh` update? Did you pull it from upstream? If it changed upstream, then we definitely want it updated, but as a separate PR. Also need to check if there were changes in any other versions upstream.

File: mysql-8.0/docker-entrypoint.sh (hunk 1 above)
Comment (sergey-zabolotny) [reply mysql-8.0/docker-entrypoint.sh]: with old docker-entrypoint.sh mysql server could not start

File: tests/test.bats (hunk 2 above)
Comment (sergey-zabolotny) [reply tests/test.bats]: fixed

File: tests/test.bats (hunk 3 above)
Comment (sergey-zabolotny) [reply tests/test.bats]: fixed

File: mysql-8.0/docker-entrypoint.sh (hunk 1 above)
Comment (sergey-zabolotny) [reply mysql-8.0/docker-entrypoint.sh]: custom code is restored

**Review by lmakarov — APPROVED** [review]

**Pull request merged**